RELEASES.md

```diff
@@ -205,3 +205,93 @@ To create a minor release:
 6. Create a pull request back to master with the CHANGELOG & version changes from the latest release.
    - Remove all `R:minor` labels from the pull requests that were included in the release.
    - Do not merge the backport branch into master.
+
+
+## Release Checklist
+
+The following set of steps are performed on all releases that increment
+the _minor_ version. These steps ensure that Tendermint is well tested, stable,
+and suitable for adoption by the various diverse projects that rely on Tendermint.
+
+### Feature Freeze
+
+Ahead of any minor version release of Tendermint, the software enters a two week
+'Feature Freeze'. A feature freeze means that _no_ new features or functionality
+are added to the code base. The Tendermint team instead directs all attention to
+ensuring that the existing code is stable and reliable. Broken tests are fixed,
+flakey-tests are remedied, end-to-end test failures are thoroughly diagnosed and
+all efforts of the team are aimed at improving the quality of the code. During
+this period, the upgrade harness tests are run repeatedly and a variety of
+in-house testnets are run to ensure Tendermint functions at the scale it will
+be used by application developers and node operators.
+
+### Nightly End-To-End Tests
+
+The Tendermint team maintains a set of end-to-end tests that run each night
+on the latest commit of the project. During the feature freeze, these tests are
+run nightly and must pass consistently for a release of Tendermint to be
+considered stable.
+
+### Upgrade Harness
+
+The Tendermint team is in the process of creating an upgrade harness. This
+harness will test the process of stopping an instance of Tendermint running one
+version of the software and starting up the same application running the next
+version of Tendermint. This harness will be paired with the ability to crash the
+Tendermint process at a series of pre-defined places in the execution to allow
+the team to ensure that Tendermint successfully upgrades in many
+different start-stop combinations.
```

Comment:
More optional wordsmithing suggestions:

```suggestion
The Tendermint team is creating an upgrade test harness to exercise the workflow 
of stopping an instance of Tendermint running one version of the software and starting up the same application running the next version.  To support upgrade
testing, we will add the ability to terminate the Tendermint process at specific
pre-defined points in its execution  so that we can verify upgrades work in a
representative sample of stop conditions.
```